### PR TITLE
theia-ide: Add version 1.49.100

### PR DESCRIPTION
**Pull request by goyalyashpal**
* Note: checkver returns `1.49.1`, download-url is at `1.49.100` so, used in manifest is `$version00` in hopes that future versions will follow the same pattern.

  * Version specific download link for theia (for use in `"autoupdate"`) https://projects.eclipse.org/projects/ecd.theia/downloads -> https://download.eclipse.org/theia / ide / latest / windows

  * Dealing with NSIS pkgs: examples at https://github.com/search via: `user:scoopinstaller path:bucket/*.json "\"extract_dir\": \"$P"` https://github.com/ScoopInstaller/Extras/blob/d48d63f0bf/bucket/android-messages.json#L10-L18

  * `checkver: github` from https://github.com/ScoopInstaller/Scoop/wiki/App-Manifest-Autoupdate#special-cases

Close: https://github.com/ScoopInstaller/Extras/issues/13272

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by goyalyashpal** [comment]
[hash-512]:  ""
[hash-ref]:  ""
[hash-variables]: https://github.com/ScoopInstaller/Scoop/wiki/App-Manifest-Autoupdate#hash-variables ""


todo: 
- [ ] use link which can support old versions (see in code block below)
- [ ] use [hash-sha512-base64] from provided yaml file via regex (ref: [hash-variables])
reason for: hash coming from root upstream is obv better than calculating ad-hoc
- [ ] use version from that file too? 
reason for: provides full version, so no `$version00` i.e. 00 suffix
reason against: github is the place linked from the theia-ide's homepage

```
https://download.eclipse.org/theia/ide/1.47.100/windows/TheiaIDESetup-1.47.100.exe
```


```
https://download.eclipse.org/theia/ide/latest/windows/latest.yml
```

**Comment by goyalyashpal** [comment]
uh ohw,
* it will fail on matching hash for older versions. 
* as the hash etc info is available only for latest version... 
* though, if someone want to install an older version of `theia-ide`, per can use `scoop install --skip` to skip hash validation

so, how to proceed?

just to be clear, there's no problem in matching hash of current version; as is evident with the github-actions 


<details><summary>Details</summary>
<p>

```console
$ scoop download theia-ide@1.47.1
Updating Scoop...
Updating Buckets...
Scoop was updated successfully!
INFO  Downloading 'theia-ide' (1.47.1) [64bit] from svah bucket
WARN  Given version (1.47.1) does not match manifest (1.49.1)
WARN  Attempting to generate manifest for 'theia-ide' (1.47.1)
Autoupdating theia-ide
DEBUG[1715549772.06301] [$updatedProperties] = [url hash] -> D:\UserFiles\scoop\apps\scoop\current\lib\autoupdate.ps1:465:5
  ...
DEBUG[1715549772.28056] $substitutions.$baseurl                       https://download.eclipse.org/theia/ide/1.47.100/windows
  ...
DEBUG[1715549772.32169] $hashfile_url = https://download.eclipse.org/theia/ide/latest/windows/latest.yml -> D:\UserFiles\scoop\apps\scoop\current\lib\autoupdate.ps1:206:5
Searching hash for TheiaIDESetup-1.47.100.exe in https://download.eclipse.org/theia/ide/latest/windows/latest.yml
DEBUG[1715549773.86048] $regex = sha512: ([a-zA-Z0-9+\/=]{24,88}) -> D:\UserFiles\scoop\apps\scoop\current\lib\autoupdate.ps1:60:9
Found: sha512:a88a979fb1446cf463517377331ed67c78cdb9e2dec70e5d139b1b751883e2cecb20d45f060d9a1a8b30d7fb374d98ddd089e60c071cb5d8fd35819ba06a9156 using Extract Mode
Writing updated theia-ide manifest
Starting download with aria2 ...
Download: [#1e0208 4.2MiB/210MiB(1%) CN:5 DL:228KiB ETA:15m27s] 
^C

$ scoop download theia-ide@1.48.3
INFO  Downloading 'theia-ide' (1.48.3) [64bit] from svah bucket
WARN  Given version (1.48.3) does not match manifest (1.49.1)
WARN  Attempting to generate manifest for 'theia-ide' (1.48.3)
Autoupdating theia-ide
DEBUG[1715549842.13152] [$updatedProperties] = [url hash] -> D:\UserFiles\scoop\apps\scoop\current\lib\autoupdate.ps1:465:5
  ...
DEBUG[1715549842.28746] $substitutions.$baseurl                       https://download.eclipse.org/theia/ide/1.48.300/windows
  ...
DEBUG[1715549842.3241] $hashfile_url = https://download.eclipse.org/theia/ide/latest/windows/latest.yml -> D:\UserFiles\scoop\apps\scoop\current\lib\autoupdate.ps1:206:5
Searching hash for TheiaIDESetup-1.48.300.exe in https://download.eclipse.org/theia/ide/latest/windows/latest.yml
DEBUG[1715549843.78344] $regex = sha512: ([a-zA-Z0-9+\/=]{24,88}) -> D:\UserFiles\scoop\apps\scoop\current\lib\autoupdate.ps1:60:9
Found: sha512:a88a979fb1446cf463517377331ed67c78cdb9e2dec70e5d139b1b751883e2cecb20d45f060d9a1a8b30d7fb374d98ddd089e60c071cb5d8fd35819ba06a9156 using Extract Mode
Writing updated theia-ide manifest
Starting download with aria2 ...
Download: [#96bebb 304KiB/196MiB(0%) CN:5 DL:184KiB ETA:18m12s]
^C
```


</p>
</details>

File: bucket/theia-ide.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+    "$schema": "https://raw.github.com/ScoopInstaller/Scoop/master/schema.json",
```
Comment (niheaven) [inline bucket/theia-ide.json:2]:
```suggestion
```

File: bucket/theia-ide.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    "version": "1.49.1",
+    "description": "A modern and open IDE for cloud and desktop. Theia platform based. Formerly “Theia Blueprint”.",
+    "homepage": "https://theia-ide.org/#theiaide",
+    "license": "EPL-2.0, GPL-2.0, MIT",
+    "notes": "Settings are stored in '%APPDATA%\\Theia IDE', and are not persisted by Scoop. For using explicit versions: neither hash, nor all old versions are available.",
```
Comment (niheaven) [inline bucket/theia-ide.json:7]:
```suggestion
    "notes": "Settings are stored in '%APPDATA%\\Theia IDE', and are not persisted by Scoop.",
```

File: bucket/theia-ide.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    },
+    "bin": "TheiaIDE.exe",
+    "shortcuts": [
+        [
+            "TheiaIDE.exe",
+            "TheiaIDE"
+        ]
+    ],
+    "checkver": {
+        "github": "https://github.com/eclipse-theia/theia"
```
Comment (niheaven) [inline bucket/theia-ide.json:28]:
```suggestion
        "github": "https://github.com/eclipse-theia/theia",
        "replace": "${1}00"
```

File: bucket/theia-ide.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://download.eclipse.org/theia/ide/$version00/windows/TheiaIDESetup-$version00.exe#/dl.7z",
```
Comment (niheaven) [inline bucket/theia-ide.json:33]:
```suggestion
                "url": "https://download.eclipse.org/theia/ide/$version/windows/TheiaIDESetup-$version.exe#/dl.7z",
```

File: bucket/theia-ide.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        "64bit": {
+            "url": "https://download.eclipse.org/theia/ide/1.49.100/windows/TheiaIDESetup-1.49.100.exe#/dl.7z",
+            "hash": "sha512:a88a979fb1446cf463517377331ed67c78cdb9e2dec70e5d139b1b751883e2cecb20d45f060d9a1a8b30d7fb374d98ddd089e60c071cb5d8fd35819ba06a9156"
+        }
+    },
+    "extract_dir": "$PLUGINSDIR",
+    "installer": {
+        "script": [
+            "Expand-7zipArchive \"$dir\\app-64.7z\" \"$dir\" -Removal"
+        ]
```
Comment (niheaven) [inline bucket/theia-ide.json:18]:
```suggestion
        "script": "Expand-7zipArchive \"$dir\\app-64.7z\" \"$dir\" -Removal"
```

File: bucket/theia-ide.json [hunk 6]
```diff
@@ -0,0 +1,41 @@
+{
+    "$schema": "https://raw.github.com/ScoopInstaller/Scoop/master/schema.json",
+    "version": "1.49.1",
```
Comment (niheaven) [inline bucket/theia-ide.json:3]:
```suggestion
    "version": "1.49.100",
```

File: bucket/theia-ide.json (hunk 1 above)
Comment (niheaven) [reply bucket/theia-ide.json:2]: Not needed

**Review by niheaven — CHANGES_REQUESTED** [review]

File: bucket/theia-ide.json (hunk 6 above)
Comment (niheaven) [reply bucket/theia-ide.json:3]: In `About` page, the version is 1.49.100.

File: bucket/theia-ide.json (hunk 3 above)
Comment (goyalyashpal) [reply bucket/theia-ide.json:28]: awesome. scoop is really well made 😄

File: bucket/theia-ide.json (hunk 6 above)
Comment (niheaven) [reply bucket/theia-ide.json:3]: The theia project is version 1.49.1 which is a framework, when came to IDE, the version is 1.49.100.

Comment (goyalyashpal) [reply bucket/theia-ide.json:3]: ohww ohkay. got it.

**Comment by goyalyashpal** [comment]
oops, added `.` at end by mistake. should i `commit --amend` it?

**Comment by niheaven** [comment]
Never mind.

File: bucket/theia-ide.json (hunk 6 above)
Comment (goyalyashpal) [reply bucket/theia-ide.json]: hey so, should checkver use `latest.yml` file ?

File: bucket/theia-ide.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        ]
+    ],
+    "checkver": {
+        "github": "https://github.com/eclipse-theia/theia",
+        "replace": "${1}00"
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://download.eclipse.org/theia/ide/$version/windows/TheiaIDESetup-$version.exe#/dl.7z",
```
Comment (goyalyashpal) [inline bucket/theia-ide.json:31]:
```suggestion
                "url": "https://www.eclipse.org/downloads/download.php?file=/theia/ide/$version/windows/TheiaIDESetup-$version.exe&r=1#/dl.7z",
```

* this download link is (inspired from) official download link on theia webpage
* it resolves the mirror itself
* remove the `&r=1` to view the download page & select the mirror manually


for me, the link resolved was this following one, which was excruciatingly slow when used with `checkhashes.ps1`: 

```xml
<mirror url="https://mirror.kakao.com/eclipse/theia/ide/latest/windows/TheiaIDESetup-1.49.100.exe" label="[Korea, Republic Of] Kakao Corp. (https)"/>
```

but the one i used above works super fast. so, i ain't sure if the automatic mirror resolution should be used or not...

```url
https://download.eclipse.org/theia/ide/latest/windows/TheiaIDESetup-1.49.100.exe
```

it is the landing/redirected location from the link listed on downloads page under the `mirror_id=1` i.e. `Main Download Site` `Canada - [Main eclipse.org downloads area]` (following link):

```url
https://www.eclipse.org/downloads/download.php?file=/theia/ide/latest/windows/TheiaIDESetup-1.49.100.exe&mirror_id=1
```

File: bucket/theia-ide.json [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    "bin": "TheiaIDE.exe",
+    "shortcuts": [
+        [
+            "TheiaIDE.exe",
+            "TheiaIDE"
+        ]
+    ],
+    "checkver": {
+        "github": "https://github.com/eclipse-theia/theia",
+        "replace": "${1}00"
```
Comment (goyalyashpal) [inline bucket/theia-ide.json:26]:
```suggestion
        "url": "https://download.eclipse.org/theia/ide/latest/windows/latest.yml",
        "regex": "version: ([\\d\\.]+)"
```

File: bucket/theia-ide.json [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        ]
+    ],
+    "checkver": {
+        "url": "https://download.eclipse.org/theia/ide/latest/windows/latest.yml",
+        "regex": "version: ([\\d\\.]+)"
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://download.eclipse.org/theia/ide/$version/windows/TheiaIDESetup-$version.exe#/dl.7z",
```
Comment (goyalyashpal) [inline bucket/theia-ide.json:31]:
```suggestion
                "url": "https://www.eclipse.org/downloads/download.php?mirror_id=1&file=/theia/ide/$version/windows/TheiaIDESetup-$version.exe#/dl.7z",
```

File: bucket/theia-ide.json [hunk 10]
```diff
@@ -0,0 +1,39 @@
+{
+    "version": "1.49.100",
+    "description": "A modern and open IDE for cloud and desktop. Theia platform based. Formerly “Theia Blueprint”.",
+    "homepage": "https://theia-ide.org/#theiaide",
+    "license": "EPL-2.0, GPL-2.0, MIT",
+    "notes": "Settings are stored in '%APPDATA%\\Theia IDE', and are not persisted by Scoop.",
+    "architecture": {
+        "64bit": {
+            "url": "https://download.eclipse.org/theia/ide/1.49.100/windows/TheiaIDESetup-1.49.100.exe#/dl.7z",
```
Comment (goyalyashpal) [inline bucket/theia-ide.json:9]:
```suggestion
            "url": "https://www.eclipse.org/downloads/download.php?mirror_id=1&file=/theia/ide/1.49.100/windows/TheiaIDESetup-1.49.100.exe#/dl.7z",
```

File: bucket/theia-ide.json [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    "bin": "TheiaIDE.exe",
+    "shortcuts": [
+        [
+            "TheiaIDE.exe",
+            "TheiaIDE"
+        ]
+    ],
+    "checkver": {
+        "url": "https://download.eclipse.org/theia/ide/latest/windows/latest.yml",
+        "regex": "version: ([\\d\\.]+)"
```
Comment (niheaven) [inline bucket/theia-ide.json:26]:
```suggestion
        "regex": "version: ([\\d.]+)"
```

In regular expressions, a dot enclosed in brackets, '.', represents a literal dot and does not require a preceding '\\'.

**Review by niheaven — APPROVED** [review]

**Comment by goyalyashpal** [comment]
holly molly. 
* soooo glad we used the `latest//latest.yml` file for `checkver`. 
* the version `1.49.101` landed here 😂😅 : https://github.com/ScoopInstaller/Extras/commit/7a3e41d0aa822b68a7d34a00ab9bea98840189a9
* the github repo is still at [1.49.1](https://github.com/eclipse-theia/theia/releases/tag/v1.49.1)


thanks for this enlightening comment 😇🪔:

> The theia project is version 1.49.1 which is a framework, when came to IDE, the version is 1.49.100.
> _\- @ niheaven at https://github.com/ScoopInstaller/Extras/pull/13274#discussion_r1601445570_

**Comment by goyalyashpal** [comment]
[eclipse-theia/theia-ide]: https://github.com/eclipse-theia/theia-ide
[eclipse-theia/theia]: https://github.com/eclipse-theia/theia
[release tags]: https://github.com/eclipse-theia/theia-ide/tags

Just for my own records,
- the source code for theia-ide is hosted & distributed at following repo: [eclipse-theia/theia-ide] · [Release tags]
- NOT the [eclipse-theia/theia] which as niheaven pointed is for theia-platform.

the main loophole by theia website is that they never linked the github repo for ide, but i suppose they would be having their own reasons as well.



<details><summary>Details</summary>
<p>

> i wonder why have they created such a discrepency in their githubb releases then?
when literally everywhere else it's *.x00 then why *.x on github 🤔
>
> \- @ me at https://github.com/ScoopInstaller/Extras/pull/13274#discussion_r1601430977


> The theia project is version 1.49.1 which is a framework, when came to IDE, the version is 1.49.100.
>
> \- @ niheaven at https://github.com/ScoopInstaller/Extras/pull/13274#discussion_r1601430977

</p>
</details>

**Pull request merged**